airflow/cli/commands/celery_command.py

```diff
@@ -96,6 +100,27 @@ def _serve_logs(skip_serve_logs: bool = False):
         sub_proc.terminate()
 
 
+@after_setup_logger.connect()
+def logger_setup_handler(logger, **kwargs):
+    # Setup levels at which logs go to stderr and stdout if required
+    if conf.get("logging", "celery_logging_split", fallback=None):
+        celery_formatter = logging.Formatter(DEFAULT_TASK_LOG_FMT)
+
+        class NoErrorOrAboveFilter(logging.Filter):
+            def filter(self, record):
+                return record.levelno <= logging.WARNING
+
+        below_error_handler = logging.StreamHandler(sys.stdout)
+        below_error_handler.addFilter(NoErrorOrAboveFilter())
+        below_error_handler.setFormatter(celery_formatter)
+
+        from_error_handler = logging.StreamHandler(sys.stderr)
+        from_error_handler.setLevel(logging.ERROR)
+        from_error_handler.setFormatter(celery_formatter)
+
+        logger.handlers[:] = [below_error_handler, from_error_handler]
```

Comment:
I wonder if it’s a good idea to replace the handlers entirely. Is it possible to have the logger already pointing to somewhere else e.g. a file?

Reply:
I was also thinking about it.
I don't think following solution is very clean, but if we want we could do:

- go through all previous handlers
- remove any StreamHandler, because we do not want duplicate logs
- leave any FileHandlers, if they are present then user probably introduced them purposefully 

Alternatively, if you think it is better I would suggest to put more weight on the fact that those handlers are cleared in documentation and docstring, so user selecting this flag will expect that only those two handlers will be present.

Reply:
Hmm yeah the workaround doesn’t seem very reliable either. Maybe we can rename the config to something more specific (that describes the exact “error to stderr, lower to file” behaviour) and describe it in the documentation.

Reply:
Maybe`celery_logging_reset_and_split`?
I also updated docstring and config.yml so it highlights a bit more the fact that those previous handlers will get removed. Do you think that it is sufficient?

Reply:
Not sure about the config, but yeah the rest looks good to me.

Reply:
Do you have an intuition what might be missing, or badly explained?
Should we have another opinion on config topic?
Let me know if I can try to fix it.

Reply:
Mostly “reset and split” seems vague to me—what does this reset logging to and split it into? I don’t have a concrete suggestion but feel like there should be a better name.

Reply:
Another proposition:

`celery_logging_remove_default_handlers_and_use_custom`

However, it is quite long.

Reply:
@uranusjr 
What do you think?

Reply:
Maybe just `celery_stdout_stderr_separation` ?

Reply:
WDYT @uranusjr ?

Reply:
IMHO `celery_stdout_stderr_separation` express the spirit of this option.
As far as I know (I contacted @uranusjr) he is OOO for at least a week more.

Reply:
let's go ahead wth this then - we can always fix it later.